### DISCUSSION TITLE
PurgeCSS misses Tailwind classes that use a colon:

### DISCUSSION THREAD

**Discussion by koaning** [tailwindlabs/tailwindcss#3774] [Help]
I have a website that uses tailwind (https://calmcode.io). I'd like to keep the css size small, so I decided to use PurgeCss. I'm building the static site with python, so I'd decided to use the CLI to process the large css file. Here's the code in my Makefile that handles the purging:

```
NODE_ENV=production npx purgecss --css css/*.css --content public/*.html public/**/*.html --output demo
npx uglifycss demo/*.css > public/style.css
rm -rf demo/*.css
```

When I run this, the strangest thing happens, it removes too many classes! It seems to remove all classes that use a color `:`. You can see the effect it has on the grid in the examples below. 

<details>
  <summary><b>What it is supposed to look like.</b></summary>
  <img src="https://user-images.githubusercontent.com/1019791/111594371-44f5c380-87cb-11eb-9cbd-8a26c20ea3a1.png"/>
</details>

<details>
  <summary><b>What it ends up looking like.</b></summary>
    <img src="https://user-images.githubusercontent.com/1019791/111594407-4f17c200-87cb-11eb-81d7-8a833dfe0a83.png"/>
</details>

Here's a snippet of the HTML that I am using (it explains the grid in the images).

```html
<ul class="grid grid-cols-1 gap-6 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 pt-4">
   ...
</ul>
```

`Purgecss` ignores the `sm:grid-cols-2`, `md:grid-cols-3` and `lg:grid-cols-4` classes which means that everything is rendered as if everyone is on mobile. I can also confirm manually that these classes do not end up in `public/style.css`.

I've also tried using the tailwind tools described [here](https://tailwindcss.com/docs/optimizing-for-production#removing-unused-css). Here's the contents of my `tailwind.config.js` file. 

```
module.exports = {
  purge: [
    'public/*.html',
    'public/**/*.html'
  ],
  darkMode: false,
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [],
}
```

Even when I run this config, it yields the same result. 

```
NODE_ENV=production npx tailwindcss-cli@latest build css/tailwind.min.css -c tailwind.config.js -o public/style.css
```

I'm wondering what the culprit is here, it feels like I'm following all the right steps but at the same time it also feels like I am completely missing something. Would anybody know of some pointers of things that I should check out?

**Comment by gwillem** [comment]
Solution here: https://stackoverflow.com/questions/65554596/purgecss-and-tailwind-css-how-to-preserve-responsive-classes-using-the-command
